### ISSUE TITLE
Next.js v15 forcefully dropping Click ID query parameters from URLs like ?fbclid / ?gclid

### ISSUE THREAD

**Issue by LoopsGod** [vercel/next.js#72102]
### Link to the code that reproduces this issue

https://codesandbox.io/p/devbox/strange-chandrasekhar-j59xzg?file=%2Fapp%2Fapi%2Fdemo%2Froute.ts&workspaceId=7095e953-179e-4203-beb8-355f8e3d65b4

### To Reproduce

The new Next.js version (V15) seems to perform an internal redirect or some other logic causing it to drop common click ID parameters from URLs. This also happens on API routes. It does not happen with `utm_` or any other parameters except what seem to be common click identification parameters. I suspect this was done in order to fix caching issues occuring because click IDs are commonly accidentally persisted. 

1. Perform a request to a Next.js V15 site with a click parameter attached to the url, e.g.: 
```/?fbclid=IwAR2F4-dbP0l7Mn1IawQQGCINEz7PYXQvwjNwB_qa2ofrHyiLjcbCRxTDMgk ``` 
2. See the 307 redirect happen.
3. Be surprised for the absence of the query parameter in `searchParams`.

## Locally 

<img width="868" alt="Screenshot 2024-10-31 at 03 41 18" src="https://github.com/user-attachments/assets/fbb45671-9f1c-47d7-8174-d0ff10f79e23">

<img width="354" alt="Screenshot 2024-10-31 at 03 44 48" src="https://github.com/user-attachments/assets/546fceb0-da51-41c9-b40c-b0db22c2e124">

<img width="354" alt="Screenshot 2024-10-31 at 03 45 15" src="https://github.com/user-attachments/assets/db72b401-4760-4351-8776-8e24f76c5c5e">

## CodeSandBox
<img width="781" alt="Screenshot 2024-10-31 at 04 02 29" src="https://github.com/user-attachments/assets/006717d5-379d-4537-8640-9dc62162d2e8">

 
<img width="672" alt="Screenshot 2024-10-31 at 04 02 39" src="https://github.com/user-attachments/assets/28c550be-2c9d-4ec3-8fad-9a8f4075b410">

### Current vs. Expected behavior

The expected behavior would be to not have odd modifications of the `searchParams`, especially not in API routes. These Click ID parameters are very important to attribute conversions accurately towards ads and for server-side Conversion and Event APIs.

Stripping of the parameters is OK, as long as the `searchParams` object will contain the value of the passed query parameters. On my V14 Next.js website, this behavior does not occur.

### Provide environment information

```bash
Operating System:
  Platform: darwin
  Arch: x64
  Version: Darwin Kernel Version 23.1.0: Mon Oct  9 21:27:27 PDT 2023; root:xnu-10002.41.9~6/RELEASE_X86_64
  Available memory (MB): 16384
  Available CPU cores: 16
Binaries:
  Node: 18.20.0
  npm: 10.5.0
  Yarn: 1.22.5
  pnpm: 9.12.1
Relevant Packages:
  next: 15.0.0-canary.152 // There is a newer canary version (15.0.3-canary.1) available, please upgrade! 
  eslint-config-next: 14.2.5
  react: 19.0.0-rc-7771d3a7-20240827
  react-dom: 19.0.0-rc-7771d3a7-20240827
  typescript: 5.5.4
```

### Which area(s) are affected? (Select all that apply)

Navigation

### Which stage(s) are affected? (Select all that apply)

next dev (local), next build (local), next start (local), Vercel (Deployed), Other (Deployed)

### Additional context

I've also reproduced on many other websites who seem to be using Next.js V15, and they all seem to exhibit stripping of major click IDs. I am interested to know how they perform advanced event matching against conversions without using click parameters. On my V14 Next.js website, this behavior does not occur.

**Comment by LoopsGod** [comment]
I just verified that this problem is not happening when using Safari. Very odd behavior.

**Comment by LoopsGod** [comment]
Wow! I am lost for words. It turns out that my Adblock Extension ublock origin performs this stripping of query parameters automatically before it even hit the server. This is why certain tracking parameters are discarded. I thought my extension was disabled. I compared against other websites where it was also enabled and by pure coincidence with one website with Next.js 14 that had it disabled. 

Solved. Note to future user: don't be like me. Disable adblock on development.
